gb-cpu/src/cpu.rs

```diff
@@ -1,5 +1,62 @@
 use crate::registers::Registers;
+use gb_bus::{AddressBus, Bus};
+use gb_clock::Ticker;
+use gb_roms::{Opcode, OpcodeGenerator};
 
+#[derive(Default)]
 pub struct Cpu {
-    _registers: Registers,
+    registers: Registers,
+}
+
+impl Cpu {
```

Comment:
no implementation of `iter() -> CpuIterator` ?

Reply:
`CpuIterator` require a ref. to `AddressBus` so the method would be in the form of `iter(addr_bus: &mut AddressBus) -> CpuIterator` and differ from the implementation we see in the stdlib (because here, `iter` will take an argument).
The name `cpu_iter` may be better, what do you think ?

Reply:
`cpu_iter(addrs_bus: &mutAddressBus -> CpuIterator` could be an option but requiring a mutable reference for a non mutable iterator seems counter-intuitive

Reply:
You're right `CpuIter.` don't need a mutable ref since It only read the address bus and not writing to it

Reply:
In this case
`cpu_iter(addrs_bus: &AddressBus -> CpuIterator`
`cpu_iter_mut(addrs_bus: &mut AddressBus -> CpuIterator`

Reply:
`CpuIter.` is only use for decoding the `opcode` and don't need to write to the address bus any data